src/core/weights.js

```diff
@@ -48,19 +57,68 @@ export function copy(w: Weights): Weights {
   };
 }
 
-/**
- * Merge multiple Weights together.
+/** Merge multiple Weights together.
+ *
+ * The resultant Weights will have every weight specified by each of the input
+ * weights.
+ *
+ * When there are overlaps (i.e. the same address is present in two or more of
+ * the Weights), then the appropriate resolver will be invoked to resolve the
+ * conflict. The resolver takes two weights and combines them to return a new
+ * weight.
  *
- * The resultant Weights will have every weight specified by each of the
- * input weights. If there are any overlaps (i.e. the same address is present
- * in two or more of the input weights), an error will be thrown. In the future,
- * we will likely modify this function to add a resolver that determines how to
- * combine multiple overlapping weights.
+ * When no resolvers are explicitly provided, merge defaults to
+ * conservative "error on conflict" resolvers.
  */
-export function merge(ws: $ReadOnlyArray<Weights>): Weights {
-  const nodeWeights = MapUtil.merge(ws.map((x) => x.nodeWeights));
-  const edgeWeights = MapUtil.merge(ws.map((x) => x.edgeWeights));
-  return {nodeWeights, edgeWeights};
+export function merge(
+  ws: $ReadOnlyArray<Weights>,
+  resolvers: ?{|+nodeResolver: NodeOperator, +edgeResolver: EdgeOperator|}
```

Comment:
This is fine; it’d also be fine to allow explicitly setting a node
resolver but not an edge resolver (or vice versa), which imho is a bit
more natural.

Reply:
The pattern that we use in some places that I think would make sense here:

Have a type where the options are all required (such as `TimelineCredParameters`).
And take optional arguments as `key?: $Shape<T>`.

Because it allows a Flow check to know we have a default value for everything.

```js
(overrides?: $Shape<T>) => {
  const defaults: T = {/* ... */};
  const guaranteedCompleteOptions: T = {...defaults, ...overrides};
};
```